flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/sink/writer/PulsarWriter.java

```diff
@@ -225,7 +225,7 @@ private TypedMessageBuilder<?> createMessageBuilder(
         } else {
             // Set default message timestamp if flink has provided one.
             Long timestamp = context.timestamp();
-            if (timestamp != null) {
+            if (timestamp != null && timestamp > 0) {
```

Comment:
Cool, this should be a bugfix for Pulsar sink.

Reply:
After reading the original implementation of `Context`. I start to wonder whether flink would provide a negative timestamp?

Reply:
Flink would provide a 0 timestamp when using `INSERT INTO topic1 SELECT FROM topic2` ;

And this would cause the write to fail.